azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,281 @@
+            return endpoints
+        if EndpointType.EventHub.value == endpoint_type.lower():
+            return endpoints.event_hubs
+        if EndpointType.ServiceBusQueue.value == endpoint_type.lower():
+            return endpoints.service_bus_queues
+        if EndpointType.ServiceBusTopic.value == endpoint_type.lower():
+            return endpoints.service_bus_topics
+        if EndpointType.CosmosDBCollection.value == endpoint_type.lower():
+            return endpoints.cosmos_db_sql_collections
+        if EndpointType.AzureStorageContainer.value == endpoint_type.lower():
+            return endpoints.storage_containers
+
+    def delete(self, endpoint_name: Optional[str] = None, endpoint_type: Optional[str] = None):
+        endpoints = self.hub_resource.properties.routing.endpoints
+        if endpoint_type:
+            if EndpointType.EventHub.value == endpoint_type.lower():
+                endpoints.event_hubs = []
+            elif EndpointType.ServiceBusQueue.value == endpoint_type.lower():
+                endpoints.service_bus_queues = []
+            elif EndpointType.ServiceBusTopic.value == endpoint_type.lower():
+                endpoints.service_bus_topics = []
+            elif EndpointType.CosmosDBCollection.value == endpoint_type.lower():
+                endpoints.cosmos_db_sql_collections = []
+            elif EndpointType.AzureStorageContainer.value == endpoint_type.lower():
+                endpoints.storage_containers = []
+
+        if endpoint_name:
+            if any(e.name.lower() == endpoint_name.lower() for e in endpoints.event_hubs):
+                remaining_endpoints = [e for e in endpoints.event_hubs if e.name.lower() != endpoint_name.lower()]
+                endpoints.event_hubs = remaining_endpoints
+            elif any(e.name.lower() == endpoint_name.lower() for e in endpoints.service_bus_queues):
+                remaining_endpoints = [e for e in endpoints.service_bus_queues if e.name.lower() != endpoint_name.lower()]
+                endpoints.service_bus_queues = remaining_endpoints
+            elif any(e.name.lower() == endpoint_name.lower() for e in endpoints.service_bus_topics):
+                remaining_endpoints = [e for e in endpoints.service_bus_topics if e.name.lower() != endpoint_name.lower()]
+                endpoints.service_bus_topics = remaining_endpoints
+            elif any(e.name.lower() == endpoint_name.lower() for e in endpoints.cosmos_db_sql_collections):
+                remaining_endpoints = [e for e in endpoints.cosmos_db_sql_collections if e.name.lower() != endpoint_name.lower()]
+                endpoints.cosmos_db_sql_collections = remaining_endpoints
+            elif any(e.name.lower() == endpoint_name.lower() for e in endpoints.storage_containers):
+                remaining_endpoints = [e for e in endpoints.storage_containers if e.name.lower() != endpoint_name.lower()]
+                endpoints.storage_containers = remaining_endpoints
```

Comment:
I think the `any` checks combined with the `elif`s here are to make sure we don't delete multiple endpoints that happen to have the same name - but I don't believe IoT Hub allows you to have multiple endpoints with the same name:

![image](https://user-images.githubusercontent.com/13545962/181842120-b1e25d37-b283-40fa-80e7-599fb2efc50e.png)

![image](https://user-images.githubusercontent.com/13545962/181842134-9108ea09-1bda-460e-aa14-bd1dae23dfdb.png)


In that case, we might be able to save a bunch of logic here by just changing it to:
```suggestion
            endpoints.event_hubs = [e for e in endpoints.event_hubs if e.name.lower() != endpoint_name.lower()]
            endpoints.service_bus_queues =  [e for e in endpoints.service_bus_queues if e.name.lower() != endpoint_name.lower()]
            endpoints.service_bus_topics = [e for e in endpoints.service_bus_topics if e.name.lower() != endpoint_name.lower()]
            endpoints.cosmos_db_sql_collections = [e for e in endpoints.cosmos_db_sql_collections if e.name.lower() != endpoint_name.lower()]
            endpoints.storage_containers = [e for e in endpoints.storage_containers if e.name.lower() != endpoint_name.lower()]
```

Reply:
will do so - will keep the if statements in case someone provides a name and type